### PR TITLE
5.x - cleanup debugger css classes

### PR DESCRIPTION
**Pull request by LordSimal**
Currently we have a mix of CSS classes present inside our Debugger: `.cake-debug` and `.cake-dbg`

This PR cleans this up so that we always use `.cake-debug` because its clearer.

I also formatted the HTML according to our code style (4 spaces, not 2), thats why that file has so many changes.

I used the following PHP file to test some of the edge cases with the PHP native webserver:
```php
<?php

use Cake\Controller\Controller;
use Cake\Core\Configure;
use Cake\Error\Debugger;
use Cake\Http\ServerRequest;

require 'vendor/autoload.php';

// Configs needed for some of the classes bellow
Configure::write('debug', true);
Configure::write('App.encoding', 'UTF-8');

echo Debugger::exportVar(true);
echo Debugger::exportVar("String");
echo Debugger::exportVar(12345);
echo Debugger::exportVar(12345.1234);
echo Debugger::exportVar(['somestring',true,3.14159265359]);
echo Debugger::exportVar(['assoc_array' => 1, 'test' => 2]);

$obj = new stdClass();
$obj->int = 1;
$obj->float = 1.2345;
$obj->string = "Holadrio";
$obj->bool = false;
echo Debugger::exportVar($obj);

$Controller = new Controller(new ServerRequest());
$Controller->viewBuilder()->setHelpers(['Html', 'Form']);
$View = $Controller->createView();

echo Debugger::exportVar($View);
```

This looks something like that:
image

### CODE REVIEW

File: src/Error/Debug/dumpHeader.html [hunk 1]
```diff
@@ -1,275 +1,284 @@
...
-.cake-dbg-string {
-  color: var(--color-green);
-  white-space: pre-wrap;
-}
-.cake-dbg-number {
-  font-weight: bold;
-  color: var(--color-blue);
-}
-.cake-dbg-const {
-  color: var(--color-yellow);
```
Comment (LordSimal) [inline src/Error/Debug/dumpHeader.html:114]: `--color-yellow` wasn't even defined but now we use the `--color-orange` (which had no usage) for constants and booleans (as you see in the screenshot above)

File: src/Error/Debug/dumpHeader.html [hunk 2]
```diff
@@ -1,275 +1,284 @@
...
+    .cake-debug-special {
+        color: var(--color-red);
+        font-style: italic;
+    }
+</style>
+<script>
+    ( function( win, doc ) {
+
+        function initialize() {
+            createCollapsibles( doc.querySelectorAll( '.cake-debug-array-items' ) );
```
Comment (markstory) [inline src/Error/Debug/dumpHeader.html]: Did you use a formatter to add spaces around the method calls? I don't remember these being here before.

**Review by markstory — COMMENTED** [review]
Changes generally look good, but the formatting in the JS code looks unlike the JS code formatting we typically use.

File: src/Error/Debug/dumpHeader.html (hunk 2 above)
Comment (LordSimal) [reply src/Error/Debug/dumpHeader.html]: I did use a formatter but it seems I used the wrong preset. let me revert that.

File: src/Error/Debugger.php [hunk 3]
```diff
@@ -793,7 +793,7 @@ public static function printVar(mixed $var, array $location = [], ?bool $showHtm
     public static function formatHtmlMessage(string $message): string
     {
         $message = h($message);
-        $message = preg_replace('/`([^`]+)`/', '<code>$1</code>', $message);
+        $message = preg_replace('/`([^`]+)`/', '<code>`$1`</code>', $message);
```
Comment (ADmad) [inline src/Error/Debugger.php]:
```suggestion
        $message = preg_replace('/`([^`]+)`/', '<code>$0</code>', $message);
```
The matches already contains string with backticks. Earlier I had explicitly showed the backticks to help you understand :)

File: tests/TestCase/Error/DebuggerTest.php [hunk 4]
```diff
@@ -732,14 +732,14 @@ public function testPrintVar(): void
...
+        $this->assertSame('Some <code>`code`</code> to <code>`replace`</code>', $output);
 
         $output = Debugger::formatHtmlMessage("Some `co\nde` to `replace`\nmore");
-        $this->assertSame("Some <code>co<br />\nde</code> to <code>replace</code><br />\nmore", $output);
+        $this->assertSame("Some <code>`co<br />\nde`</code> to <code>`replace`</code><br />\nmore", $output);
 
         $output = Debugger::formatHtmlMessage("Some `code` to <script>alert(\"test\")</script>\nmore");
         $this->assertSame(
-            "Some <code>code</code> to &lt;script&gt;alert(&quot;test&quot;)&lt;/script&gt;<br />\nmore",
+            "Some <code>`code`</code> to &lt;script&gt;alert(&quot;test&quot;)&lt;/script&gt;<br />\nmore",
```
Comment (markstory) [inline tests/TestCase/Error/DebuggerTest.php:742]: ~~I don't think we need the backticks as the HTML will be formatted as inline code.~~

Disregard this, I read the pull requests in a reverse order.

**Comment by LordSimal** [comment]
Can this be merged or is there something missing?
